### PR TITLE
Create book UI

### PR DESCRIPTION
**Pull request by barbaravaldez**
Initial work for adding the create book dialog. 
The inputs of the dialog are the name of the book, the save location and an optional content folder. 
If the content folder is undefined, then we create an "empty" book (I actually need to add an empty readme, so it can be opened in the Books View). If the content folder contains files, then I create the toc based on the contents.
The create book command still opens the create jupyter notebook.

Screen Shot 2021-02-09 at 12 19 18 PM

### CODE REVIEW

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -142,10 +143,10 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		}
 	}
 
-	async createBook(bookPath: string, contentPath: string): Promise<void> {
-		bookPath = path.normalize(bookPath);
-		contentPath = path.normalize(contentPath);
-		await this.bookTocManager.createBook(bookPath, contentPath);
+	async createBook(): Promise<void> {
+		const bookTocManager = new BookTocManager();
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:147]: Do we really want/need to create a new toc manager every time we do this? It looks like we're just using this to create a book so seems like we can just have a single one we re-use

File: extensions/notebook/src/book/bookTocManager.ts [hunk 2]
```diff
@@ -221,14 +221,23 @@ export class BookTocManager implements IBookTocManager {
...
 	*/
-	public async createBook(bookContentPath: string, contentFolder: string): Promise<void> {
-		await fs.promises.mkdir(bookContentPath);
-		await fs.copy(contentFolder, bookContentPath);
-		let filesinDir = await fs.readdir(bookContentPath);
-		this.tableofContents = await this.getAllFiles([], bookContentPath, filesinDir, bookContentPath);
+	public async createBook(bookContentPath: string, contentFolder?: string): Promise<void> {
+		let filesinDir: string[];
+		await fs.promises.mkdir(bookContentPath, { recursive: true });
+		if (!!contentFolder) {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTocManager.ts:232]: You don't need the `!!` - if the value is truthy it'll go in this block regardless

File: extensions/notebook/src/book/bookTocManager.ts [hunk 3]
```diff
@@ -221,14 +221,23 @@ export class BookTocManager implements IBookTocManager {
...
+			await fs.copy(contentFolder, bookContentPath);
+			filesinDir = await fs.readdir(bookContentPath);
+			this.tableofContents = await this.getAllFiles([], bookContentPath, filesinDir, bookContentPath);
+		} else {
+			await fs.writeFile(path.join(bookContentPath, 'readme.md'), '');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTocManager.ts:237]: `README` is usually capitalized

**Review by Charles-Gagnon — APPROVED** [review]
Please always include screenshots when adding/modifying UI elements

File: extensions/notebook/src/book/bookTocManager.ts [hunk 4]
```diff
@@ -221,14 +221,23 @@ export class BookTocManager implements IBookTocManager {
 	/**
 	 * Follows the same logic as the JupyterBooksCreate.ipynb. It receives a path that contains a notebooks and
 	 * a path where it creates the book. It copies the contents from one folder to another and creates a table of contents.
-	 * @param bookContentPath The path to the book folder, the basename of the path is the name of the book
-	 * @param contentFolder The path to the folder that contains the notebooks and markdown files to be added to the created book.
+	 * @param bookContentPath - The path to the book folder, the basename of the path is the name of the book
+	 * @param contentFolder - (Optional) The path to the folder that contains the notebooks and markdown files to be added to the created book.
+	 * If it's undefined then a blank notebook is attached to the book.
+	 *
```
Comment (lucyzhang929) [inline extensions/notebook/src/book/bookTocManager.ts:227]: nit: extra line

**Review by lucyzhang929 — APPROVED** [review]

**Pull request merged**